backends/gdx-backend-moe/src/com/badlogic/gdx/backends/iosmoe/IOSPreferences.java

```diff
@@ -193,5 +195,7 @@ public void run() {
 				}
 			}
 		});
+		success = true; // FIXME: should reflect the success of NSDictionary write.

```

Comment:
You could change line 188 to `final boolean[] success = {false};`, success set on line 194 with `success[0] = true;` and `return success[0];`. This pattern is often used in similar situations. It leads to an allocation, but if the MOE is smart enough, it could optimize it out. And if not - flush is a slow operation anyway, so it doesn't matter that much.